.github/scripts/prep_release.py

```diff
@@ -9,15 +9,21 @@
     (pytest's script I stole a lot from)
 """
 
+from __future__ import annotations
+
 import argparse
-import datetime
+import datetime as dt
 import pathlib
 import re
+import shutil
 import subprocess
 from textwrap import indent
+from typing import TYPE_CHECKING
 
 import github3
-from github3.repos import Repository
+
+if TYPE_CHECKING:
+    from github3.repos import Repository
 
```

Comment:
Just curious about the general use here. Is the `TYPE_CHECKING` check in this case to make it clear that `Repository` is only used by static type checkers, or to speed up the code (ever so slightly?) when running normally?

Reply:
Yeah, exactly. It delineates runtime imports vs imports solely for type checking which does add a (minor) performance boost, but also helps prevent circular imports in some cases. I think it's a good habit/practice.